modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -197,13 +219,16 @@ private boolean assertConsistencyBetweenHttpRequestAndOperation(Request<?> reque
         }
     }
 
-    private static long getRequestCount(Request<?> request) {
-        Number requestCount = request.getAWSRequestMetrics().getTimingInfo().getCounter(AWSRequestMetrics.Field.RequestCount.name());
-        if (requestCount == null) {
-            logger.warn("Expected request count to be tracked for request [{}] but found not count.", request);
+    private static long getCountForMetric(TimingInfo info, AWSRequestMetrics.Field field) {
+        var count = info.getCounter(field.name());
+        if (count == null) {
+            if (field == AWSRequestMetrics.Field.RequestCount) {
+                logger.warn("Expected request count to be tracked but found not count.");
+            }
```

Comment:
Is this log still relevant? Should it be replaced with assertion?

Reply:
Or may be we should have a metric for request count without such counter?

Reply:
I added an assertion, but left the logging to be. I don't think it is needed. I can do a follow-up refactoring to maybe remove it as part of the effort to consolidate all the APM instruments into its own class.

Reply:
Thanks for digging into this. I think this metric field being missing indicating a bug in either our side or the s3 client side. But since we don't control s3 client, it might be worth to leave the logging as is for now.